### ISSUE TITLE
Exponential decay reset after resuming training

### ISSUE THREAD

**Issue by AlexFuster** [magenta/ddsp#60]
Once the training is stopped and launched again, it continues training from the last checkpoint, however, the optimizer schedule is always reinicialized.
This happens in the \_\_init\_\_ method of the class Trainer in ddsp/training/train_util.py
```python    
    lr_schedule = tf.keras.optimizers.schedules.ExponentialDecay(
        initial_learning_rate=learning_rate,
        decay_steps=lr_decay_steps,
        decay_rate=lr_decay_rate)

    with self.strategy.scope():
      optimizer = tf.keras.optimizers.Adam(lr_schedule)
      self.optimizer = optimizer
```
A new instance of Trainer is created each time ddsp_run is executed

**Comment by AlexFuster** [comment]
I may be wrong with this, as I see that optimizer is included in the training checkpoint:
```python
checkpoint = tf.train.Checkpoint(model=self.model, optimizer=self.optimizer)
```
give me your thoughts about this

**Comment by jesseengel** [comment]
Yah, I believe that it's doing the right thing as the trainer loading loads the optimizer state (which I believe includes learning rate decay).

**Comment by AlexFuster** [comment]
yeah, you're right, I close this one
